experimental/packages/opentelemetry-sdk-metrics-base/src/aggregator/Sum.ts

```diff
@@ -37,29 +44,43 @@ export class SumAccumulation implements Accumulation {
 export class SumAggregator implements Aggregator<SumAccumulation> {
   public kind: AggregatorKind.SUM = AggregatorKind.SUM;
 
-  createAccumulation() {
-    return new SumAccumulation();
+  constructor (public monotonic: boolean) {}
+
+  createAccumulation(startTime: HrTime) {
+    return new SumAccumulation(startTime, this.monotonic);
   }
 
   /**
    * Returns the result of the merge of the given accumulations.
    */
   merge(previous: SumAccumulation, delta: SumAccumulation): SumAccumulation {
-    return new SumAccumulation(previous.toPointValue() + delta.toPointValue());
+    if (delta.reset) {
+      return new SumAccumulation(delta.startTime, this.monotonic, delta.toPointValue(), delta.reset);
+    }
+    return new SumAccumulation(previous.startTime, this.monotonic, previous.toPointValue() + delta.toPointValue());
   }
 
   /**
    * Returns a new DELTA aggregation by comparing two cumulative measurements.
    */
   diff(previous: SumAccumulation, current: SumAccumulation): SumAccumulation {
-    return new SumAccumulation(current.toPointValue() - previous.toPointValue());
+    const prevPv = previous.toPointValue();
+    const currPv = current.toPointValue();
+    /**
+     * If the SumAggregator is a monotonic one and the previous point value is
+     * greater than the current one, a reset is deemed to be happened.
+     * Return the current point value to prevent the value from been reset.
+     */
+    if (this.monotonic && (prevPv > currPv)) {
+      return new SumAccumulation(current.startTime, this.monotonic, currPv, true);
+    }
+    return new SumAccumulation(current.startTime, this.monotonic, current.toPointValue() - previous.toPointValue());
   }
 
   toMetricData(
     descriptor: InstrumentDescriptor,
     aggregationTemporality: AggregationTemporality,
     accumulationByAttributes: AccumulationRecord<SumAccumulation>[],
-    startTime: HrTime,
     endTime: HrTime): Maybe<SingularMetricData> {
     return {
       descriptor,
```

Comment:
I'm wondering if we should indicate `isMonotonic` in `SingularMetricData`. We have access to the data anyway, it is part of the proto, and it simplifies the transformation to OTLP. 

No need to do this here (it would probably blow up the diff too much), but might be worth to do it in a follow-up. :slightly_smiling_face:

Reply:
Yeah, I'd agree this is a large PR already and we can update the OTLP transformation in a follow-up PR.

Reply:
Sounds good. I can pick this up in a follow up. :slightly_smiling_face: